### PR TITLE
Add Equals and HashCode to Request and Response Classes

### PR DESCRIPTION
**Pull request by poojabhat0**
## Description

1. Add `equals()` and `hashCode()` methods to public Request and Response classes
2. Add Builders for all public classes
3. Add Javadoc for Builder methods
4. Add unit tests for testing `equals()` and `hashCode()` 

## Related Issues
1. https://github.com/tecton-ai/tecton-http-client-java/issues/47
2. https://github.com/tecton-ai/tecton-http-client-java/issues/48

### CODE REVIEW

File: src/main/java/ai/tecton/client/model/ListDataType.java [hunk 1]
```diff
@@ -38,4 +40,23 @@ class ListDataType {
...
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    ListDataType that = (ListDataType) o;
+    return Objects.equals(stringList, that.stringList)
+        && Objects.equals(float32List, that.float32List)
+        && Objects.equals(float64List, that.float64List)
+        && Objects.equals(int64List, that.int64List)
+        && listElementType == that.listElementType;
```
Comment (achals) [inline src/main/java/ai/tecton/client/model/ListDataType.java:54]: Why not have Objects.equals here too?

File: src/main/java/ai/tecton/client/model/NameAndType.java [hunk 2]
```diff
@@ -58,4 +60,21 @@ public ValueType getDataType() {
...
+
+  /** Overrides <i>equals()</i> in class {@link Object} */
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    NameAndType that = (NameAndType) o;
+    return Objects.equals(name, that.name)
+        && dataType == that.dataType
+        && listElementType == that.listElementType;
```
Comment (achals) [inline src/main/java/ai/tecton/client/model/NameAndType.java:72]: Same question here

File: src/main/java/ai/tecton/client/TectonClient.java [hunk 3]
```diff
@@ -188,4 +188,85 @@ private HttpResponse getHttpResponse(AbstractTectonRequest tectonRequest) {
...
+    private TectonClientOptions tectonClientOptions;
+    private OkHttpClient httpClient;
+
+    /**
+     * Setter for url
+     *
+     * @param url The Tecton Base Url
+     * @return this Builder
+     */
+    public Builder url(String url) {
```
Comment (achals) [inline src/main/java/ai/tecton/client/TectonClient.java:208]: Id check but does the kotlin assignment operator need a specific method name? Like 'Builder.setX' which kotlin can then translate to 'builder.X = something'?

**Review by achals — APPROVED** [review]

File: src/main/java/ai/tecton/client/model/ListDataType.java (hunk 1 above)
Comment (poojabhat0) [reply src/main/java/ai/tecton/client/model/ListDataType.java:54]: The listElementType is an enum

File: src/main/java/ai/tecton/client/TectonClient.java (hunk 3 above)
Comment (poojabhat0) [reply src/main/java/ai/tecton/client/TectonClient.java:208]: Tested in a Kotlin Application and this works

**Pull request merged**